packages/flutter/lib/src/widgets/navigator.dart

```diff
@@ -769,6 +769,13 @@ class NavigatorObserver {
   /// The [Navigator] replaced `oldRoute` with `newRoute`.
   void didReplace({ Route<dynamic>? newRoute, Route<dynamic>? oldRoute }) { }
 
+  /// The top most route has changed.
+  ///
+  /// The `topRoute` is the new top most route. This can be a new route pushed
+  /// on top of the screen, or an existing route that becomes the new top-most
+  /// route because the previous top-most route has been popped.
+  void didChangeTop(Route<dynamic> topRoute, Route<dynamic>? previousTopRoute) { }
```

Comment:
Can you add some documentation about previousTopRoute? Probably worthwhile to mention that this could be a route that is about to be disposed (if it was popped off) or continues to live on if it was covered by a new route. Also, when is it null?